tokio-util/tests/sync_cancellation_token.rs

```diff
@@ -493,3 +493,51 @@ fn run_until_cancelled_test() {
         );
     }
 }
+
+#[test]
+fn run_until_cancelled_owned_test() {
```

Comment:
I think it would be nice to add a case to this test that would only compile with the `run_until_cancelled_owned` and not with `run_until_cancelled` like you mentioned in the PR description as this was the reason behind this PR.